src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -524,6 +524,9 @@ class RsErrorAnnotator : AnnotatorBase(), HighlightRangeExtension {
             !impl.isUnsafe && !trait.isUnsafe && impl.excl == null && attrRequiringUnsafeImpl != null ->
                 RsDiagnostic.TraitMissingUnsafeImplAttributeError(traitRef, attrRequiringUnsafeImpl).addToHolder(holder)
         }
+
+        val const = impl.const ?: return
+        CONST_TRAIT_IMPL.check(holder, const, "const trait impls")
```

Comment:
```suggestion
        val const = impl.const
        if (const != null) {
            CONST_TRAIT_IMPL.check(holder, const, "const trait impls")
        }
```

Let's avoid return in complex checks because it may affects all subsequent checks.
For example, `val traitRef = impl.traitRef ?: return` above prevent you from checking inherent impls with `const`